docs/automated-testing/unit-testing/mocking.md

```diff
@@ -92,16 +92,128 @@ since it is held to the same standard with full unit tests. Especially if writin
 developers can use, providing a fake in the product code means those developers no longer need to write their own mock
 implementations, further increasing re-usability of code.
 
-### Upsides
+### Upsides to Fakes
 
 - No framework needed, is just like any other implementation.
 - Encourages testable design.
 - Code can be "promoted" to product code, so it is not wasted effort.
 
-### Downsides
+### Downsides to Fakes
 
 - Takes more time to implement.
 
+## Best Practices
```

Comment:
The need for dependency injection to do mocking is only relevant for certain languages (ex. C#) but many languages (ex. Python) doesn't need this, nor is it a common practice to do dependency injection. Might be worth mentioning this.

Reply:
Thanks for the feedback! I added a disclaimer about different languages :)